examples/md-flexible/Simulation.h

```diff
@@ -0,0 +1,469 @@
+/**
+ * @file Simulation.h
+ * @author N. Fottner
+ * @date 12/05/19
+ */
+#pragma once
+
+#include <fstream>
+#include <iostream>
+
+#include "../../tests/testAutopas/testingHelpers/GaussianGenerator.h"
+#include "../../tests/testAutopas/testingHelpers/GridGenerator.h"
+#include "../../tests/testAutopas/testingHelpers/RandomGenerator.h"
+#include "BoundaryConditions.h"
+#include "Checkpoint.h"
+#include "Generator.h"
+#include "PrintableMolecule.h"
+#include "Thermostat.h"
+#include "TimeDiscretization.h"
+#include "autopas/AutoPas.h"
+#include "autopas/molecularDynamics/LJFunctorAVX.h"
+#include "autopas/utils/MemoryProfiler.h"
+#include "parsing/MDFlexConfig.h"
+
+template <class Particle, class ParticleCell>
+class Simulation {
+ public:
+  /**
+   * Constructor.
+   *
+   * This starts the timer for total simulation time.
+   */
+  explicit Simulation() { _timers.total.start(); };
+
+  /**
+   * Destructor.
+   *
+   * Closes the log file if applicable.
+   */
+  ~Simulation() {
+    if (not _config->logFileName.empty()) {
+      _logFile.close();
+    }
+  }
+
+  /**
+   * Writes a VTK file for the current state of the AutoPas object
+   * @tparam AutoPasTemplate Template for the templetized autopas type.
+   * @param filename
+   * @param numParticles
+   * @param autopas
+   */
+  void writeVTKFile(unsigned int iteration) {
+    _timers.vtk.start();
+
+    std::string fileBaseName = _config->vtkFileName;
+    // as _autopas.getNumberOfParticles return number of haloAndOwned Particles, we need number of owned Particles
```

Comment:
if this is true, why do you use getNumberOfParticles() then?

Reply:
!

Reply:
-> sensible function will be built in #377